modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -291,7 +291,10 @@ const renderSplitButton = (spec: Toolbar.ToolbarSplitButton, sharedBackstage: Ui
     },
 
     onExecute: (button: AlloyComponent) => {
-      spec.onAction(getApi(button));
+      const api = getApi(button);
+      if (api.isEnabled()) {
```

Comment:
I wonder why regular buttons don't need this check. I can only assume that their use of events for `onExecute` means the execute event doesn't fire when the button is disabled, because disabled runs first and blocks everything else:

https://github.com/tinymce/tinymce/blob/3a6455aaebf6034e122ef6cf86beeecef2c7103b/modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ButtonEvents.ts#L30

I'm sure there's an interesting story about why split buttons don't handle execute in the same way. Oh well.

Reply:
I believe it is connected to calling an action explicitly from `SplitDropdownSchema`. I can be wrong though.

https://github.com/tinymce/tinymce/blob/5d1cf51eac3184b29c92b17686712fbcc9f5f736/modules/alloy/src/main/ts/ephox/alloy/ui/schema/SplitDropdownSchema.ts#L95-L99